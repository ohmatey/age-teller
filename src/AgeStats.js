import React from 'react'
import { compose, withPropsOnChange, branch, renderNothing, renderComponent } from 'recompose'

import calcDateDifference from './utils/calcDateDifference'

export const AgeStats = ({ date, diffedDate }) => (
  <div>
    <h2>{date}</h2>
    <h5>{diffedDate}</h5>
  </div>
)

const enhance = compose(
  withPropsOnChange(
    (props, nextProps) => props.date !== nextProps.date,
    ({ date }) => ({
      diffedDate: calcDateDifference(date)
    })
  )
)

export const enhancedAgeStats = enhance(AgeStats)

const HasDateAgeStats = branch(
  ({ date }) => !date,
  renderNothing,
  renderComponent(enhancedAgeStats)
)

export default HasDateAgeStats()
