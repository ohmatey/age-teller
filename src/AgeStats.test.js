import React from 'react'
import ReactDOM from 'react-dom'
import renderer from 'react-test-renderer'

import { AgeStats } from './AgeStats'

test('AgeStats renders without props', () => {
  const component = renderer.create(
    <AgeStats />
  )

  const tree = component.toJSON()
  expect(tree).toMatchSnapshot()
})

test('AgeStats renders with date', () => {
  const mockDate = '22-12-2014'
  const component = renderer.create(
    <AgeStats date={mockDate} />
  )

  const tree = component.toJSON()
  expect(tree).toMatchSnapshot()
})

test('AgeStats renders with diffedDate', () => {
  const mockDate = '22-12-2014'
  const component = renderer.create(
    <AgeStats diffedDate={mockDate} />
  )

  const tree = component.toJSON()
  expect(tree).toMatchSnapshot()
})
