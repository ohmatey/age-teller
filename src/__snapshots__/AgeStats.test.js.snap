// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AgeStats renders with date 1`] = `
<div>
  <h2>
    22-12-2014
  </h2>
</div>
`;

exports[`AgeStats renders with diffedDate 1`] = `
<div>
  <h2 />
</div>
`;

exports[`AgeStats renders without props 1`] = `
<div>
  <h2 />
</div>
`;
